### ISSUE TITLE
Support YARN_PRODUCTION=false on Yarn 2 to skip pruning

### ISSUE THREAD

**Issue by snake575** [heroku/heroku-buildpack-nodejs#906]
Hi, I started having errors on my build since #891 added support for pruning dev dependencies on Yarn 2, it's a great feature but the `YARN_PRODUCTION` flag still isn't supported to skip pruning. On my case setting it to `false` results on the following error:

```
-----> Installing dependencies
       Running 'yarn install' with yarn.lock
       Usage Error: Unrecognized or legacy configuration settings found: production - run "yarn config -v" to see the list of settings supported in Yarn (in <environment>)

       $ yarn install [--json] [--immutable] [--immutable-cache] [--check-cache] [--inline-builds] [--skip-builds]
```

**Comment by danielleadams** [comment]
The PR you're referring to had actually not worked until I merged https://github.com/heroku/heroku-buildpack-nodejs/pull/905 on Friday (2/26). Can you give me more details? Yarn version? Node version? Exactly what day the build started failing? Thanks.

**Comment by snake575** [comment]
Sorry, I noticed the build failing on Saturday 2/27 02:59 UTC, using `v183` worked (I haven't tested again since using the main branch).

These were the binaries:

node: `14.16.0`
npm: `6.14.11`  
yarn: `1.22.10`
yarn2: `2.4.0`

Runtime environment:
```
NPM_CONFIG_LOGLEVEL=error
USE_YARN_CACHE=true
NODE_ENV=production
NODE_MODULES_CACHE=true
NODE_VERBOSE=false
```

**Comment by osdiab** [comment]
I also experienced this, you can repro it locally by just running `YARN_PRODUCTION=false yarn install`.

Probably need to choose a different environment variable name for skipping pruning, this one breaks `yarn install`.

**Comment by osdiab** [comment]
Also I think the tests look wrong? shouldn't this be `assertNotCaptured`?

https://github.com/heroku/heroku-buildpack-nodejs/blob/main/test/run#L944

**Comment by osdiab** [comment]
`YARN2_PRODUCTION` could work though.

**Comment by rachel-church** [comment]
Running into the same issue. +1 vote for @osdiab's suggestion to add a more broad `SKIP_PRUNE_DEV_DEPENDENCIES` flag

**Comment by hyperparabolic** [comment]
+1, please add a toggle for this feature.

I use the node-linker in a monorepo where I only deploy one workspace.  I check in a local cache that I validate as a part of an external CI process, and I have postbuild steps that:
1) `yarn workspaces focus some_workspace --production`
2) delete my cache after that install step.

Which has made my heroku builds very fast and resilient up until this point.

This "prune" step is actually increasing my slug size, has increased my deploy time by several minutes, and makes my build process less consistent.

**Comment by kobajs** [comment]
Same issue here

**Comment by oleksandr-danylchenko** [comment]
Any updates on this issue? 
Still throwing an exception in the Heroku: `Usage Error: Unrecognized or legacy configuration settings found: production - run "yarn config -v" to see the list of settings supported in Yarn (in <environment>)` when I set `YARN_PRODUCTION=false`

**Comment by martinbee** [comment]
Same issue here.

**Comment by mdesantis** [comment]
Same issue here. Any workaround? I can't prune dev dependencies as they are required for assets compiling

**Comment by colincasey** [comment]
this should be addressed once #1008 is released
